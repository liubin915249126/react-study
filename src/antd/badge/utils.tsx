import { PresetColorTypes } from '../_util/colors';

// eslint-disable-next-line import/prefer-default-export
export function isPresetColor(color?: string): boolean {
  return (PresetColorTypes as any[]).indexOf(color) !== -1;
}
