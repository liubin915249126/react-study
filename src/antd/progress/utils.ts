import devWarning from '../_util/devWarning';

export function validProgress(progress: number | undefined) {
  if (!progress || progress < 0) {
    return 0;
  }
  if (progress > 100) {
    return 100;
  }
  return progress;
}

export function getSuccessPercent({
  success,
  successPercent,
}: {
  success?: {
    progress?: number;
    percent?: number;
  };
  successPercent?: number;
}) {
  let percent = successPercent;
  /** @deprecated Use `percent` instead */
  if (success && 'progress' in success) {
    devWarning(
      false,
      'Progress',
      '`success.progress` is deprecated. Please use `success.percent` instead.',
    );
    percent = success.progress;
  }
  if (success && 'percent' in success) {
    percent = success.percent;
  }
  return percent;
}
