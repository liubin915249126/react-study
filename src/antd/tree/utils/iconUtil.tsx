import * as React from 'react';
import classNames from 'classnames';
import LoadingOutlined from '@ant-design/icons/LoadingOutlined';
import FileOutlined from '@ant-design/icons/FileOutlined';
import MinusSquareOutlined from '@ant-design/icons/MinusSquareOutlined';
import PlusSquareOutlined from '@ant-design/icons/PlusSquareOutlined';
import CaretDownFilled from '@ant-design/icons/CaretDownFilled';
import { AntTreeNodeProps } from '../Tree';
import { isValidElement, cloneElement } from '../../_util/reactNode';

export default function renderSwitcherIcon(
  prefixCls: string,
  switcherIcon: React.ReactNode | null | undefined,
  showLine: boolean | { showLeafIcon: boolean } | undefined,
  { isLeaf, expanded, loading }: AntTreeNodeProps,
) {
  if (loading) {
    return <LoadingOutlined className={`${prefixCls}-switcher-loading-icon`} />;
  }
  let showLeafIcon;
  if (showLine && typeof showLine === 'object') {
    showLeafIcon = showLine.showLeafIcon;
  }
  if (isLeaf) {
    if (showLine) {
      if (typeof showLine === 'object' && !showLeafIcon) {
        return <span className={`${prefixCls}-switcher-leaf-line`} />;
      }
      return <FileOutlined className={`${prefixCls}-switcher-line-icon`} />;
    }
    return null;
  }
  const switcherCls = `${prefixCls}-switcher-icon`;
  if (isValidElement(switcherIcon)) {
    return cloneElement(switcherIcon, {
      className: classNames(switcherIcon.props.className || '', switcherCls),
    });
  }

  if (switcherIcon) {
    return switcherIcon;
  }

  if (showLine) {
    return expanded ? (
      <MinusSquareOutlined className={`${prefixCls}-switcher-line-icon`} />
    ) : (
      <PlusSquareOutlined className={`${prefixCls}-switcher-line-icon`} />
    );
  }
  return <CaretDownFilled className={switcherCls} />;
}
