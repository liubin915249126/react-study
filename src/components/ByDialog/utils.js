export const openFun = ({callback,destroy}) => {
    if (!!callback
        && (typeof callback === 'object' || typeof callback === 'function')) {
        const promiseResult = callback() || {};
        if (typeof promiseResult.then === 'function') {
            promiseResult.finally(() => {
                destroy();
            });
        } else {
            destroy();
        }
    } else {
        destroy();
    }
}