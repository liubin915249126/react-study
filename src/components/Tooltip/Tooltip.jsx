import React from 'react';

class Tooltip extends React.Component{
    constructor(props){
        super(props)
    }
    componentDidMount(){
        console.log(this.props)
    }
    componentWillReceiveProps(nextProps){
       
    }
    render(){
        return this.props.children
    }
}
export default Tooltip;