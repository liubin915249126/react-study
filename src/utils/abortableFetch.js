import { AbortController, AbortSignal } from "../lib/abort-controller";

const abortControllers = []

if (typeof Symbol !== "undefined" && Symbol.toStringTag) {
  // These are necessary to make sure that we get correct output for:
  // Object.prototype.toString.call(new AbortController())
  AbortController.prototype[Symbol.toStringTag] = "AbortController";
  AbortSignal.prototype[Symbol.toStringTag] = "AbortSignal";
}
const realFetch = fetch;
const abortableFetch = (input, init) => {
  console.log('input',input)
  const abortController = new AbortController();
  abortControllers.push(abortController)
  // console.log(2222,abortController,abortControllers)
  if (abortController) {
    const abortError = new Error("Aborted");
    abortError.name = "AbortError";
    abortError.isAborted = true;

    // Return early if already aborted, thus avoiding making an HTTP request
    if (abortController.aborted) {
      return Promise.reject(abortError);
    }
    // Turn an event into a promise, reject it once `abort` is dispatched
    const cancellation = new Promise((resolve, reject) => {
      abortController.signal.addEventListener(
        "abort",
        () => {
          console.log(1111,'abort')
          resolve({
            status:"401"
          });
        },
        { once: true }
      );
    });

    // delete init.signal;

    // Return the fastest promise (don't need to wait for request to finish)
    return Promise.race([cancellation, realFetch(input, init)]).then((res)=>{
      console.log('res',res)
      return Promise.resolve(res)
    }).catch((err)=>{
      console.log('err',err)
    });
  }

  // return realFetch(input, init);
};

export { abortableFetch, abortControllers };
