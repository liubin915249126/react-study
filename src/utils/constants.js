export const POPOVER_ROOT = 'popover-root';
export const MODAL_ROOT = 'modal-root';

export const TEXT_ALIGNS = {
  LEFT: undefined,
  CENTER: 'center',
  RIGHT: 'right',
};

export const COMPONENT_TYPES = {
  DEFAULT: undefined,
  OUTLINED: 'outlined',
  CONTAINED: 'contained',
};

export const SIZE_ENUMS = {
  NONE: undefined,
  MEDIUM: 'medium',
  SMALL: 'small',
  LARGE: 'large',
  XLARGE: 'x-large',
  XXLARGE: 'xx-large',
};

export const COLOR_ENUMS = {
  DEFAULT: undefined,
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  LONG: 'long',
  SHORT: 'short',
  BRAND: 'brand',
  WHITE: 'white',
  GC1: 'gc-1',
  GC2: 'gc-2',
  GC4: 'gc-4',
};

export const TRIGGER = {
  CLICK: 'click',
  HOVER: 'hover',
};

export const INFO_TYPES = {
  ERROR: 'error',
  WARN: 'warn',
  SUCCESS: 'success',
  INFO: 'info',
};

const placementBase = ['top', 'bottom', 'left', 'right'];
const placementModifier = ['-start', '-end'];
/**
 * Generate placement constants by above definitions, combined 1 by 1.
 * result: {
 *  AUTO: 'auto',
 *  TOP: 'top',
 *  TOP_START: 'top-start',
 *  TOP_END: 'top-end',
 *  // etc...
 * }
 */
export const PLACEMENT = {
  AUTO: 'auto',
  ...placementBase.reduce((prev, cur) => ({
    ...prev,
    ...placementModifier.reduce(
      (pre, cu) => ({
        ...pre,
        [cur.toUpperCase()]: cur,
        [`${cur}${cu}`.toUpperCase().replace('-', '_')]: `${cur}${cu}`,
      }),
      {},
    ),
  }), {}),
};
