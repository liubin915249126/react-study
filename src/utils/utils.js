import moment from 'moment';

export function fixedZero(val) {
  return val * 1 < 10 ? `0${val}` : val;
}

export function getTimeDistance(type) {
  const now = new Date();
  const oneDay = 1000 * 60 * 60 * 24;

  if (type === 'today') {
    now.setHours(0);
    now.setMinutes(0);
    now.setSeconds(0);
    return [moment(now), moment(now.getTime() + (oneDay - 1000))];
  }

  if (type === 'week') {
    let day = now.getDay();
    now.setHours(0);
    now.setMinutes(0);
    now.setSeconds(0);

    if (day === 0) {
      day = 6;
    } else {
      day -= 1;
    }

    const beginTime = now.getTime() - (day * oneDay);

    return [moment(beginTime), moment(beginTime + ((7 * oneDay) - 1000))];
  }

  if (type === 'month') {
    const year = now.getFullYear();
    const month = now.getMonth();
    const nextDate = moment(now).add(1, 'months');
    const nextYear = nextDate.year();
    const nextMonth = nextDate.month();

    return [moment(`${year}-${fixedZero(month + 1)}-01 00:00:00`), moment(moment(`${nextYear}-${fixedZero(nextMonth + 1)}-01 00:00:00`).valueOf() - 1000)];
  }

  if (type === 'year') {
    const year = now.getFullYear();

    return [moment(`${year}-01-01 00:00:00`), moment(`${year}-12-31 23:59:59`)];
  }
}

export function getPlainNode(nodeList, parentPath = '') {
  const arr = [];
  nodeList.forEach((node) => {
    const item = node;
    item.path = `${parentPath}/${item.path || ''}`.replace(/\/+/g, '/');
    item.exact = true;
    if (item.children && !item.component) {
      arr.push(...getPlainNode(item.children, item.path));
    } else {
      if (item.children && item.component) {
        item.exact = false;
      }
      arr.push(item);
    }
  });
  return arr;
}

export function digitUppercase(n) {
  const fraction = ['角', '分'];
  const digit = ['零', '壹', '贰', '叁', '肆', '伍', '陆', '柒', '捌', '玖'];
  const unit = [
    ['元', '万', '亿'],
    ['', '拾', '佰', '仟'],
  ];
  let num = Math.abs(n);
  let s = '';
  fraction.forEach((item, index) => {
    s += (digit[Math.floor(num * 10 * (10 ** index)) % 10] + item).replace(/零./, '');
  });
  s = s || '整';
  num = Math.floor(num);
  for (let i = 0; i < unit[0].length && num > 0; i += 1) {
    let p = '';
    for (let j = 0; j < unit[1].length && num > 0; j += 1) {
      p = digit[num % 10] + unit[1][j] + p;
      num = Math.floor(num / 10);
    }
    s = p.replace(/(零.)*零$/, '').replace(/^$/, '零') + unit[0][i] + s;
  }

  return s.replace(/(零.)*零元/, '元').replace(/(零.)+/g, '零').replace(/^整$/, '零元整');
}


function getRelation(str1, str2) {
  if (str1 === str2) {
    console.warn('Two path are equal!');  // eslint-disable-line
  }
  const arr1 = str1.split('/');
  const arr2 = str2.split('/');
  if (arr2.every((item, index) => item === arr1[index])) {
    return 1;
  } else if (arr1.every((item, index) => item === arr2[index])) {
    return 2;
  }
  return 3;
}

function getRenderArr(routes) {
  let renderArr = [];
  renderArr.push(routes[0]);
  for (let i = 1; i < routes.length; i += 1) {
    let isAdd = false;
    // 是否包含
    isAdd = renderArr.every(item => getRelation(item, routes[i]) === 3);
    // 去重
    renderArr = renderArr.filter(item => getRelation(item, routes[i]) !== 1);
    if (isAdd) {
      renderArr.push(routes[i]);
    }
  }
  return renderArr;
}

/**
 * Get router routing configuration
 * { path:{name,...param}}=>Array<{name,path ...param}>
 * @param {string} path
 * @param {routerData} routerData
 */
export function getRoutes(path, routerData) {
  let routes = Object.keys(routerData).filter(routePath =>
    routePath.indexOf(path) === 0 && routePath !== path);
  // Replace path to '' eg. path='user' /user/name => name
  routes = routes.map(item => item.replace(path, ''));
  // Get the route to be rendered to remove the deep rendering
  const renderArr = getRenderArr(routes);
  // Conversion and stitching parameters
  const renderRoutes = renderArr.map((item) => {
    const exact = !routes.some(route => route !== item && getRelation(route, item) === 1);
    return {
      ...routerData[`${path}${item}`],
      key: `${path}${item}`,
      path: `${path}${item}`,
      exact,
    };
  });
  return renderRoutes;
}


/* eslint no-useless-escape:0 */
const reg = /(((^https?:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)$/g;

export function isUrl(path) {
  return reg.test(path);
}

export function randStr(len) {
  return Math.random().toString(36).substr(-len);
}

/**
 * deepclone
 */
export function deepClone(source) {
  if (!source && typeof source !== 'object') {
    throw new Error('error arguments', 'shallowClone');
  }
  const targetObj = source.constructor === Array ? [] : {};
  for (const keys in source) {
    if (Object.prototype.hasOwnProperty.call(source, keys)) {
      if (source[keys] && typeof source[keys] === 'object') {
        targetObj[keys] = source[keys].constructor === Array ? [] : {};
        targetObj[keys] = deepClone(source[keys]);
      } else {
        targetObj[keys] = source[keys];
      }
    }
  }
  return targetObj;
}
export function clone(item) {
  if (!item) { return item; } // null, undefined values check
  const types = [Number, String, Boolean];
  let result;

  // normalizing primitives if someone did new String('aaa'), or new Number('444');
  types.forEach((type) => {
    if (item instanceof type) {
      result=type(item);
    }
  });

  if (typeof result === 'undefined') {
    if (Object.prototype.toString.call(item) === '[object Array]') {
      result = [];
      item.forEach((child, index) => {
        result[index] = clone(child);
      });
    } else if (typeof item === 'object') {
      // testing that this is DOM
      if (item.nodeType && typeof item.cloneNode === 'function') {
        let result = item.cloneNode(true);
      } else if (!item.prototype) { // check that this is a literal
        if (item instanceof Date) {
          result = new Date(item);
        } else {
          // it is an object literal
          result = {};
          for (const i in item) {
            if (item[i]) {
              result[i] = clone(item[i]);
            }
          }
        }
      } else if (false && item.constructor) {
        // depending what you would like here,
        // just keep the reference, or create new object
        // would not advice to do that, reason? Read below
        result = new item.constructor();
      } else {
        result = item;
      }
    } else {
      result = item;
    }
  }

  return result;
}
export function isObjectValueEqual(a, b) {
  if (typeof a === 'number' && typeof b === 'number') {
    return a === b;
  }
  if (a && b) {
    const aProps = Object.getOwnPropertyNames(a);
    const bProps = Object.getOwnPropertyNames(b);
    if (aProps.length !== bProps.length) {
      return false;
    }
    for (let i = 0; i < aProps.length; i += 1) {
      const propName = aProps[i];
      if (Object.prototype.toString(a[propName]) === '[Object Object]' || Object.prototype.toString(b[propName]) === '[Object Object]') {
        isObjectValueEqual(a[propName], b[propName]);
      }
      if (a[propName] !== b[propName]) {
        return false;
      }
    }
  } else {
    return false;
  }
  return true;
}
