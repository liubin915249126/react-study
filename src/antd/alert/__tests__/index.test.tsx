import React from 'react';
import { mount } from 'enzyme';
import Alert from '..';
import Tooltip from '../../tooltip';
import Popconfirm from '../../popconfirm';
import rtlTest from '../../../tests/shared/rtlTest';
import { sleep } from '../../../tests/utils';

const { ErrorBoundary } = Alert;

describe('Alert', () => {
  rtlTest(Alert);

  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('could be closed', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <Alert
        message="Warning Text Warning Text Warning TextW arning Text Warning Text Warning TextWarning Text"
        type="warning"
        closable
        onClose={onClose}
      />,
    );
    wrapper.find('.ant-alert-close-icon').simulate('click');
    jest.runAllTimers();
    expect(onClose).toHaveBeenCalled();
  });

  describe('data and aria props', () => {
    it('sets data attributes on input', () => {
      const wrapper = mount(<Alert data-test="test-id" data-id="12345" message={null} />);
      const input = wrapper.find('.ant-alert').getDOMNode();
      expect(input.getAttribute('data-test')).toBe('test-id');
      expect(input.getAttribute('data-id')).toBe('12345');
    });

    it('sets aria attributes on input', () => {
      const wrapper = mount(<Alert aria-describedby="some-label" message={null} />);
      const input = wrapper.find('.ant-alert').getDOMNode();
      expect(input.getAttribute('aria-describedby')).toBe('some-label');
    });

    it('sets role attribute on input', () => {
      const wrapper = mount(<Alert role="status" message={null} />);
      const input = wrapper.find('.ant-alert').getDOMNode();
      expect(input.getAttribute('role')).toBe('status');
    });
  });

  const testIt = process.env.REACT === '15' ? it.skip : it;
  testIt('ErrorBoundary', () => {
    // @ts-expect-error
    // eslint-disable-next-line react/jsx-no-undef
    const ThrowError = () => <NotExisted />;
    const wrapper = mount(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>,
    );
    // eslint-disable-next-line jest/no-standalone-expect
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('could be used with Tooltip', async () => {
    const ref = React.createRef<any>();
    jest.useRealTimers();
    const wrapper = mount(
      <Tooltip title="xxx" mouseEnterDelay={0} ref={ref}>
        <Alert
          message="Warning Text Warning Text Warning TextW arning Text Warning Text Warning TextWarning Text"
          type="warning"
        />
      </Tooltip>,
    );
    wrapper.find('.ant-alert').simulate('mouseenter');
    await sleep(0);
    expect(ref.current.getPopupDomNode()).toBeTruthy();
    jest.useFakeTimers();
  });

  it('could be used with Popconfirm', async () => {
    const ref = React.createRef<any>();
    jest.useRealTimers();
    const wrapper = mount(
      <Popconfirm ref={ref} title="xxx">
        <Alert
          message="Warning Text Warning Text Warning TextW arning Text Warning Text Warning TextWarning Text"
          type="warning"
        />
      </Popconfirm>,
    );
    wrapper.find('.ant-alert').simulate('click');
    await sleep(0);
    expect(ref.current.getPopupDomNode()).toBeTruthy();
    jest.useFakeTimers();
  });

  it('could accept none react element icon', () => {
    const wrapper = mount(<Alert message="Success Tips" type="success" showIcon icon="icon" />);
    expect(wrapper).toMatchRenderedSnapshot();
  });
});
