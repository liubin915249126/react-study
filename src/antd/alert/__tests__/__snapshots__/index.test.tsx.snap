// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert ErrorBoundary 1`] = `
<div
  class="ant-alert ant-alert-error ant-alert-with-description ant-alert-no-icon"
  data-show="true"
  role="alert"
>
  <span
    class="ant-alert-message"
  >
    ReferenceError: NotExisted is not defined
  </span>
  <span
    class="ant-alert-description"
  >
    <pre>
          in ThrowError
    in ErrorBoundary (created by WrapperComponent)
    in WrapperComponent
    </pre>
  </span>
</div>
`;

exports[`Alert could accept none react element icon 1`] = `
<div
  class="ant-alert ant-alert-success"
  data-show="true"
  role="alert"
>
  <span
    class="ant-alert-icon"
  >
    icon
  </span>
  <span
    class="ant-alert-message"
  >
    Success Tips
  </span>
  <span
    class="ant-alert-description"
  />
</div>
`;

exports[`Alert rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-alert ant-alert-info ant-alert-no-icon ant-alert-rtl"
  data-show="true"
  role="alert"
>
  <span
    class="ant-alert-message"
  />
  <span
    class="ant-alert-description"
  />
</div>
`;
